File: xaml_masterdetail/CS/ViewModels/ItemViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using MasterDetailApp.Data;
using Windows.Globalization.DateTimeFormatting;

namespace MasterDetailApp.ViewModels
{
    public class ItemViewModel
    {
        private int _itemId;

        public int ItemId
        {
            get
            {
                return _itemId;
            }
        }

        public string DateCreatedHourMinute
        {
            get
            {
                var formatter = new DateTimeFormatter("hour minute");
                return formatter.Format(DateCreated);
            }
        }

        public string Title { get; set; }
        public string Text { get; set; }
        public DateTime DateCreated { get; set; }

        public ItemViewModel()
        {
        }

        public static ItemViewModel FromItem(Item item)
        {
            var viewModel = new ItemViewModel();

            viewModel._itemId = item.Id;
            viewModel.DateCreated = item.DateCreated;
            viewModel.Title = item.Title;
            viewModel.Text = item.Text;

            return viewModel;
        }
    }
}

File: Samples/XamlMasterDetail/cppwinrt/ItemViewModel.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "ViewModels.ItemViewModel.g.h"

namespace winrt::MasterDetailApp::ViewModels::implementation
{
    struct ItemViewModel : ItemViewModelT<ItemViewModel>
    {
        ItemViewModel(MasterDetailApp::Data::Item const& item) :
            m_id{ item.ItemId() }, m_title{ item.Title() },
            m_text{ item.Text() }, m_dateCreated{ item.DateCreated() }
        {
        }

        static MasterDetailApp::ViewModels::ItemViewModel FromItem(MasterDetailApp::Data::Item const& item)
        {
            return make<ItemViewModel>(item);
        }

        int ItemId() { return m_id; }
        hstring Title() { return m_title; }
        hstring Text() { return m_text; }
        Windows::Foundation::DateTime DateCreated() { return m_dateCreated; }

        hstring DateCreatedHourMinute()
        {
            auto formatter = Windows::Globalization::DateTimeFormatting::DateTimeFormatter(L"hour minute");
            return formatter.Format(m_dateCreated);
        }

    private:
        int const m_id;
        hstring const m_title;
        hstring m_text;
        Windows::Foundation::DateTime const m_dateCreated;
    };
}

namespace winrt::MasterDetailApp::ViewModels::factory_implementation
{
    struct ItemViewModel : ItemViewModelT<ItemViewModel, implementation::ItemViewModel>
    {
    };
}

#include "ViewModels.ItemViewModel.g.cpp"
